synapse/config/server.py

```diff
@@ -107,12 +107,23 @@ def read_config(self, config):
         federation_domain_whitelist = config.get(
             "federation_domain_whitelist", None
         )
+
         # turn the whitelist into a hash for speed of lookup
         if federation_domain_whitelist is not None:
             self.federation_domain_whitelist = {}
             for domain in federation_domain_whitelist:
                 self.federation_domain_whitelist[domain] = True
 
+        # Optional proxy address for federation traffic
+        self.federation_request_gateway_addr = config.get(
+            "federation_request_gateway_addr", None
+        )
+
+        if self.federation_request_gateway_addr is not None:
+            # Ensure proxy address is correctly formatted
+            if len(self.federation_request_gateway_addr.split(':')) != 2:
+                self.federation_request_gateway_addr = None
```

Comment:
could you throw a `ConfigError` if you don't like the value, rather than subtly and mysteriously ignoring it?